tools/Docker-dev/Dockerfile

```diff
@@ -0,0 +1,143 @@
+# docker build  --rm -f "Dockerfile" -t <build-tag> "."    
+# 
+# To use the container use the following command. It assumes that you are in
+# the root folder of the scipy git repository, making it available as
+# /home/scipy in the container. Whatever changes you make to that directory
+# are visible in the host and container.
+# The docker image is retrieved from the scipy dockerhub repository
+#
+# docker run --rm -it -v $(pwd):/home/scipy scipy:<image-tag>
+# 
+# Once the container is running, you'll need to activate the conda environment
+# conda activate scipydev
+# 
+# And run the tests with
+# python runtests.py
+# 
+# Ubuntu 20.04 (focal)
+# https://hub.docker.com/_/ubuntu/?tab=tags&name=focal
+# OS/ARCH: linux/amd64
+ARG ROOT_CONTAINER=ubuntu:focal-20201106@sha256:4e4bc990609ed865e07afc8427c30ffdddca5153fd4e82c20d8f0783a291e241
+ARG BASE_CONTAINER=${ROOT_CONTAINER}
+
+# hadolint ignore=DL3006 as base container is tagged
+FROM ${BASE_CONTAINER}
+
+# Change default shell - this avoids issues with Conda later
+SHELL ["/bin/bash","--login", "-o", "pipefail", "-c"]
+
+# -----------------------------------------------------------------------------
+# ---- Miniforge installer ----
+# Default values can be overridden at build time
+# (ARGS are in lower case to distinguish them from ENV)
+# Check https://github.com/conda-forge/miniforge/releases
+# Conda version
+ARG conda_version="4.9.2"
+# Miniforge installer patch version
+ARG miniforge_patch_number="5"
+# Miniforge installer architecture
+ARG miniforge_arch="x86_64"
+# Python implementation to use 
+# can be either Miniforge3 to use Python or Miniforge-pypy3 to use PyPy
+ARG miniforge_python="Miniforge3"
+
+# Miniforge archive to install
+ARG miniforge_version="${conda_version}-${miniforge_patch_number}"
+# Miniforge installer
+ARG miniforge_installer="${miniforge_python}-${miniforge_version}-Linux-${miniforge_arch}.sh"
+# Miniforge checksum
+ARG miniforge_checksum="49dddb3998550e40adc904dae55b0a2aeeb0bd9fc4306869cc4a600ec4b8b47c"
+
+# -----------------------------------------------------------------------------
+# ---- Python version to install ----
+# Currently Python 3.8
+ARG PYTHON_VERSION=default
+
+# ---- Configure environment ----
+ENV CONDA_DIR=/opt/conda \
+    SHELL=/bin/bash 
+
+ENV PATH=$CONDA_DIR/bin:$PATH \
+    CONDA_VERSION="${conda_version}" \
+    MINIFORGE_VERSION="${miniforge_version}" \
+    CONDA_ENV=scipydev \
+    PROJECT_DIR=/home/scipy
+
+# -----------------------------------------------------------------------------
+# ---- OS dependencies ----
+ENV DEBIAN_FRONTEND noninteractive
+
+# hadolint ignore=DL3008
+RUN apt-get update && \ 
+    apt-get install -yq --no-install-recommends \
+    software-properties-common \
+    python3.8-dev \
+    build-essential \
+    vim \
+    libatlas-base-dev \
+    gfortran \
+    libgfortran4 \
+    liblapack-dev \
+    curl \
+    libgmp-dev \
+    libmpfr-dev \
+    libsuitesparse-dev \
+    libmpc-dev \
```

Comment:
These are optional dependencies. I think it makes sense to add them here even though technically it's better to get them from conda-forge if possible. We never touch them and they do help make the tests run faster.

Reply:
It could conceivably lead to problems though, because this is also why `gfortran` is needed here, and it's ignored because there's a `gfortran` in the `compilers` conda package too.

Reply:
Right - I was nor sure which deps are **mandatory** as I looked at the Dockerfile already existing in the repo.

But I can remove anything that is not necessary and ensure conda installs superseeds these

Reply:
The [conda-forge recipe](https://github.com/conda-forge/scipy-feedstock/blob/master/recipe/meta.yaml) doesn't have any of these. They're not necessary to build, and they don't speed up the tests either unless `mpmath` is installed. So leaving all of this stuff out for now is fine I think. Maybe just add a comment in `environment.yml`:
```
# Optional dependencies left out: mpmath, gmpy2, libsuitesparse, libgmp, libmpfr, libsuitesparse, libmpc, sparse, scikit-umfpack
```